feature/preview/src/main/java/com/google/jetpackcamera/feature/preview/PreviewScreen.kt

```diff
@@ -111,7 +90,9 @@ fun PreviewScreen(
             onStopVideoRecording = viewModel::stopVideoRecording,
             onToggleCaptureMode = viewModel::toggleCaptureMode,
             onToastShown = viewModel::onToastShown,
-            onSurfaceProviderCreated = { deferredSurfaceProvider.complete(it) }
+//            onSurfaceProviderReady = { surfaceProvider = it },
+            onSurfaceProviderReady = viewModel::runCamera,
```

Comment:
We could probably clean things up just as well by adding some more code to the ViewModel so that `onSurfaceProviderReady` here would just call `viewModel::setSurfaceProvider`.

We would still need a `LifecycleStartEffect` to start/stop the camera, but the `SurfaceProvider` handling would be removed form composition. I still like the idea of starting/stopping the camera being at the `PreviewScreen` level, since we are then saying "as long as the PreviewScreen is showing, the camera should be on"